lib/src/rules/invalid_case_patterns.dart

```diff
@@ -59,7 +59,9 @@ class _Visitor extends SimpleAstVisitor {
         rule.reportLint(expression);
       }
     } else if (expression is PrefixExpression) {
-      rule.reportLint(expression);
+      if (expression.operand is! IntegerLiteral) {
```

Comment:
I'm guessing DoubleLiteral would also be affected?

Reply:
Double literals are not valid const expressions in Dart `<= 2.19` and will produce a (different) [warning](https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=case_expression_type_implements_equals#case_expression_type_implements_equals).

Reply:
Oh! Sorry.